core/src/geometry/size.rs

```diff
@@ -182,7 +182,17 @@ impl Size {
     }
 
     /// Creates a size from two corner points of a bounding box.
-    pub(crate) const fn from_bounding_box(corner_1: Point, corner_2: Point) -> Self {
+    ///
+    /// ```
+    /// use embedded_graphics::geometry::{Size, Point};
+    ///
+    /// let top_left = Point::new(5, 5);
+    /// let bottom_right = Point::new(314, 234);
+    /// let bounded = Size::from_bounding_box(top_left, bottom_right);
+    ///
+    /// assert_eq!(bounded, Size::new(310, 230));
+    /// ```
+    pub const fn from_bounding_box(corner_1: Point, corner_2: Point) -> Self {
```

Comment:
There has been some discussion about the name of this method when it was added, but at that time we didn't really care because it was internal only. I don't like the current name, because I wouldn't expect a function that is called `from_BOUNDING_BOX` take points as parameters and not a `Rectangle`. But I cannot really think of anything better at the moment and perhaps I'm overthinking this. 

Another option would be to not make this method public and instead to make `Rectangle::with_corners` const. In my opinion `Rectangle::with_corners(top_left, bottom_right).size` is more readable than `Size::from_bounding_box(top_left, bottom_right)`. And in case it is used in a const context the possible overhead from also calculating `Rectangle::top_left` doesn't matter.